BaseStation/SensorDiagnostics/SensorRealTimeGraph/MainWindow.xaml.cs

```diff
@@ -0,0 +1,150 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Input;
+
+namespace HuskyRobotics.UI
+{
+    /// <summary>
+    /// Interaction logic for MainWindow.xaml
+    /// </summary>
+    public partial class sensorRealTimeGraph : UserControl, INotifyPropertyChanged
+    {
+        private DateTime lastUpdate { get; set; }
+        private double _axisMax;
+        private double _axisMin;
+        private List<DataUnit> lastSensorReadings { get; set; }
+        public SeriesCollection seriesCollection { get; set; } // Data to be displayed
+        public SeriesCollection exampleSeriesCollection { get; set; } // Example Data used for testing
+        public Func<double, string> YFormatter { get; set; } // Yaxis format
+        public Func<double, string> DateTimeFormatter { get; set; } // Xaxis format
+        public double AxisStep { get; set; }
+        public double AxisUnit { get; set; }
+        private Random rnd = new Random();
+        /// <summary>
+        /// Constructor: Sets up ui
+        /// </summary>
+        public sensorRealTimeGraph()
+        {
+            InitializeComponent();
+            lastSensorReadings = new List<DataUnit>();
+            var mapper = Mappers.Xy<MeasureModel>()
+                .X(model => model.DateTime.Ticks)   //use DateTime.Ticks as X
+                .Y(model => model.Value);           //use the value property as Y
+            //lets save the mapper globally.
+            Charting.For<MeasureModel>(mapper);
+            YFormatter = valueY => valueY.ToString("0.##");
+            DateTimeFormatter = value => new DateTime((long)value).ToString("mm:ss");
+            //AxisStep forces the distance between each separator in the X axis
+            AxisStep = TimeSpan.FromSeconds(1).Ticks;
+            //AxisUnit forces lets the axis know that we are plotting seconds
+            //this is not always necessary, but it can prevent wrong labeling
+            AxisUnit = TimeSpan.TicksPerSecond;
+            SetAxisLimits(DateTime.Now);
+            seriesCollection = new SeriesCollection
+            {
+            };
+            DataContext = this;
+            lastUpdate = DateTime.Now;
+            updateGraph();
+        }
+
+        public void tempMethod()
+        {
+            seriesCollection[0].Values.Add(rnd.NextDouble() * 100);
+            seriesCollection[0].Values.RemoveAt(0);
+        }
+
+        public double AxisMax
+        {
+            get { return _axisMax; }
+            set
+            {
+                _axisMax = value;
+                OnPropertyChanged("AxisMax");
+            }
+        }
+        public double AxisMin
+        {
+            get { return _axisMin; }
+            set
+            {
+                _axisMin = value;
+                OnPropertyChanged("AxisMin");
+            }
+        }
+
+        private void SetAxisLimits(DateTime now)
```

Comment:
I'm confused by what this method is done. It's called SetAxisLimits, but doesn't take in any limits, instead a single argument which represents now

Reply:
It keeps the scope of the graph to a set distance behind and in front of the current time. I'll add comments to clarify some things 👍